### PR TITLE
add unit tests to run runtime migrations

### PR DESCRIPTION
**Pull request by kianenigma**
We should not replace the try-runtime testing in release process with this, as this is all executing in native and is not realistic. Nonetheless, this can now be toyed with as simple as: 

```
cargo remote -b "RUST_LOG=remote-ext=debug,runtime=debug" test -- --release -p westend-runtime -p polkadot-runtime -p kusama-runtime --features try-runtime
```

### CODE REVIEW

File: runtime/westend/src/lib.rs [hunk 1]
```diff
@@ -1174,7 +1174,10 @@ pub type Executive = frame_executive::Executive<
 	frame_system::ChainContext<Runtime>,
 	Runtime,
 	AllPalletsWithSystem,
-	(pallet_staking::migrations::v10::MigrateToV10<Runtime>,),
+	(
+		pallet_staking::migrations::v10::MigrateToV10<Runtime>,
+		pallet_nomination_pools::migration::v2::MigrateToV2<Runtime>,
+	),
```
Comment (coderobe) [inline runtime/westend/src/lib.rs:1180]:
```suggestion
	(),
```

These don't need to be here anymore

**Review by coderobe — APPROVED** [review]

File: runtime/westend/src/lib.rs [hunk 2]
```diff
@@ -1748,3 +1751,40 @@ sp_api::impl_runtime_apis! {
...
+
+	#[tokio::test]
+	async fn run_migrations() {
+		sp_tracing::try_init_simple();
+		let transport: Transport = std::option_env!("WSS")
+			.unwrap_or("wss://westend-rpc.polkadot.io:443")
+			.to_string()
+			.into();
+		let maybe_state_snapshot: Option<SnapshotConfig> =
+			std::option_env!("SNAP").map(|s| s.to_string().into());
```
Comment (bkchr) [inline runtime/westend/src/lib.rs:1771]: Why not parse the env variable at runtime? Why at compile time?

**Review by bkchr — APPROVED** [review]
:heart:

File: runtime/westend/src/lib.rs (hunk 1 above)
Comment (kianenigma) [reply runtime/westend/src/lib.rs:1180]: ah yeah I was just testing.

File: runtime/westend/src/lib.rs (hunk 2 above)
Comment (kianenigma) [reply runtime/westend/src/lib.rs:1768]: no particular preference to be honest, I supposedly had the syntax for this on the top of my head 🤷

Comment (ggwpez) [reply runtime/westend/src/lib.rs:1768]: I think its better to do at runtime, we had CI problems in the past where the CI did not provide the ENV variables.

**Review by ggwpez — APPROVED** [review]
So this allows to run the try-runtime as a rust test? Very cool!

**Comment by kianenigma** [comment]
> So this allows to run the try-runtime as a rust test? Very cool!

> We should not replace the try-runtime testing in release process with this, as this is all executing in native and is not realistic. Nonetheless, this can now be toyed with. 

I think this is fine for PR testing, but before release we should still execute stuff with wasm

File: runtime/westend/src/lib.rs [hunk 3]
```diff
@@ -1174,7 +1174,7 @@ pub type Executive = frame_executive::Executive<
 	frame_system::ChainContext<Runtime>,
 	Runtime,
 	AllPalletsWithSystem,
-	(pallet_staking::migrations::v10::MigrateToV10<Runtime>,),
+	(),
```
Comment (kianenigma) [inline runtime/westend/src/lib.rs:1177]:
```suggestion
	(pallet_staking::migrations::v10::MigrateToV10<Runtime>,),
```

**Review by shawntabrizi — APPROVED** [review]

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**